src/pages/events/createEvents/createEventCriteria/CriteriaCard.tsx

```diff
@@ -0,0 +1,312 @@
+    setNewSubFilters([
+      {
+        id: Math.random(),
+        name: '',
+        result: true,
+      },
+    ])
+  }
+
+
+  return (
+<Box>
+    <Grid columnSpacing={2} rowSpacing={2} container>
+      {params.criteriaData?.map((item: ICriteriaPaperInfo, index: number) => (
+        <Grid key={index} columns={2} item xs={6} className={classes.criteriasPaper}>
+          <Box className={classes.criteriaPaperBackground}>
+            <Box className={classes.criteriasPaperHeaderBox}>
+              <Box className={classes.checkboxManagerBox}>
+                <Checkbox
+                  checked={item.criteria}
+                  onChange={(e) => {
+                    const newObj = {...item, criteria: e.target.checked}
+                    const newArr = params.criteriaData.map((newItem: ICriteriaPaperInfo) => {
+                      if (newItem.id === newObj.id) {
+                        return newObj
+                      } else {
+                        return newItem
+                      }
+                    })
+                    params.setCriteriaData(newArr)
+                  }}
+                  className={classes.checkbox}
+                  defaultChecked
+                  icon={<UnCheckedIcon />}
+                  checkedIcon={<CheckedIcon />}
+                />
+                <Typography className={classes.criteriasPaperHeaderText}>
+                  {item.name}
+                </Typography>
+              </Box>
+              <Button
+                className={classes.criteriaAddButton}
+                onClick={(e) => handleClickCriteria(e, item.id)}
+                variant="text"
+                disableRipple
+              >
+                <Typography className={classes.criteriasAddButtonText}>Add</Typography>
+              </Button>
+            </Box>
+            <Box className={classes.criteriasPaperInfo}>
+              {item.subCriteria.map((el: ISubCriteria, index1: number) => (
+                <Box className={classes.eachCriteria} key={index1}>
+                  <Checkbox
+                    checked={el.result}
+                    onChange={(e) => {
+                      const newObj = {...el, result: e.target.checked}
+                      const newArrSubcriteria = item.subCriteria.map(
+                        (newItem: ISubCriteria) => {
+                          if (newItem.id === newObj.id) {
+                            return newObj
+                          } else {
+                            return newItem
+                          }
+                        },
+                      )
+                      const bigObj = {...item, subCriteria: newArrSubcriteria}
+                      const newBigArr = params.criteriaData.map((a: ICriteriaPaperInfo) => {
+                        if (bigObj.id === a.id) {
+                          return bigObj
+                        } else {
+                          return a
+                        }
```

Comment:
do not write a as value name, think of some variable name that will define the object

Reply:
no else needed